### ISSUE TITLE
sign-error in fanflat geometry

### ISSUE THREAD

**Issue by exargon** [astra-toolbox/astra-toolbox#132]
Hi, I have been using astra for a project and encountered some unexpected behaviour in the fanflat geometry. When I input the source_to_origin and origin_to_detector distances as positive quantities in the projection geometry the resulting reconstruction amplifies the fanbeam error (making it worse than using a parallel reconstruction on the data) instead of compensating for it.  Indeed I have to set both these quantities as negative to get the correct geometry, but then the reconstructed image results in the inverse of the expected reconstruction.

I have reproduced this error in all the algorithms I have tried so far, notably FBP_CUDA, SIRT_CUDA and SART,  which all work as expected with the parallel geometry.
 
I suspect you have a simple sign-error somewhere in your code that is causing this, but I have not been able to locate it so far.

**Comment by wjp** [comment]
This sounds like it could be a clockwise/counterclockwise rotation issue. Increasing angles correspond to counterclockwise motion in ASTRA.  Did you test what happens if you change the specified rotation direction, by for exampling flipping the sign of the angles?

**Comment by exargon** [comment]
Yeah that works :) thanks for the quick reply.  I guess the back of my mind told me this would just mirror the reconstruction, and I didn't test it. 

So as I understand it now, after thinking and trying a little bit, the error is caused by the direction of rotation not being correct compared to the direction of u in the sinogram? Because the intersection between a ray skewed towards the direction of rotation with the next projection is "pushed" away from the source and with a ray skewed in the opposite direction it is "pulled" towards the source? 

To get the desired non mirrored image(which you get when the rotation direction is changed) I can just flip the sinogram (so the leftmost detector pixel becomes the rightmost) and then shift the angles 180 degrees.

Maybe also a good idea to make it clear in the documentation that ASTRA uses the counterclockwise direction (even though this is the mathematical standard at least this makes it pop up on peoples debugging radar)
